File: client/src/game/api/emits/lg.ts
import type { GlobalId } from "../../id";
import { wrapSocket } from "../helpers";
import { socket } from "../socket";

export const sendLgTokenConnect = wrapSocket<{ typeId: number; uuid: GlobalId }>("Lg.Token.Connect");
export const sendLgShowGridIds = (): void => {
    socket.emit("Lg.Grid.Ids.Show");
};
export const sendLgHideGridIds = (): void => {
    socket.emit("Lg.Grid.Ids.Hide");
};

File: server/src/api/socket/lg.py
from ... import auth
from ...app import app, sio
from ...state.game import game_state
from ..helpers import _send_game
from .constants import GAME_NS


@sio.on("Lg.Token.Connect", namespace=GAME_NS)
@auth.login_required(app, sio, "game")
async def token_connect(sid: str, data: int):
    pr = game_state.get(sid)

    for psid in game_state.get_sids(room=pr.room, skip_sid=sid):
        await _send_game("Lg.Token.Connect", data, room=psid)


@sio.on("Lg.Grid.Ids.Show", namespace=GAME_NS)
@auth.login_required(app, sio, "game")
async def show_grid_ids(sid: str):
    pr = game_state.get(sid)

    await _send_game("Lg.Grid.Ids.Show", None, room=pr.room.get_path(), skip_sid=sid)


@sio.on("Lg.Grid.Ids.Hide", namespace=GAME_NS)
@auth.login_required(app, sio, "game")
async def hide_grid_ids(sid: str):
    pr = game_state.get(sid)

    await _send_game("Lg.Grid.Ids.Hide", None, room=pr.room.get_path(), skip_sid=sid)
    await _send_game("Lg.Grid.Ids.Hide", None, room=pr.room.get_path(), skip_sid=sid)
